go-controller/pkg/crd/userdefinednetwork/v1/types.go

```diff
@@ -0,0 +1,157 @@
+	// +required
+	Role NetworkRole `json:"role"`
+
+	// mtu is the maximum transmission unit for a network.
+	// MTU is optional, if not provided, the globally configured value in OVN-Kubernetes (defaults to 1400) is used for the network.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65536
+	// from https://github.com/openshift/api/blob/0d46442e8df17a87cfcf6666ab19b28e88620b59/network/v1/types.go#L66
+	// +optional
+	MTU int32 `json:"mtu,omitempty"`
+
+	// subnets are used for the pod network across the cluster.
+	// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+	//
+	// When topology is "Layer3", the given subnet is split into smaller subnets for every node.
+	// To specify how the subnet should be split, the following format is supported for "Layer3" network:
+	// "10.128.0.0/16/24", which means that every host will get a "/24" subnet.
+	// If host subnet mask is not set (for example, "10.128.0.0/16"), it will be assigned automatically.
+	//
+	// For "Layer2" and "LocalNet" topology types, the format should match standard CIDR notation, without
+	// providing any host subnet mask.
+	// This field may be omitted for Secondary Layer2 and LocalNet networks.
+	// In that case the logical switch implementing the network only provides layer 2 communication,
+	// and users must configure IP addresses for the pods.
+	// Port security only prevents MAC spoofing
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Subnets []string `json:"subnets,omitempty"`
+
+	// excludeSubnets is a list of CIDRs that will be removed from the assignable IP address pool specified by the "subnets" field.
+	// Should only be passed if "subnets" are set.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	// +optional
+	ExcludeSubnets []string `json:"excludeSubnets,omitempty"`
+
+	// joinSubnets are used inside the OVN network topology.
+	// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+	// This field is only allowed for "Primary" networks with "Layer2" or "Layer3" topology.
+	// It is not recommended to set this field without explicit need and understanding of the OVN network topology.
+	// When omitted, the platform will choose a reasonable default which is subject to change over time.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	JoinSubnets []string `json:"joinSubnets,omitempty"`
+
+	// ipamLifecycle controls IP addresses management lifecycle.
+	// The only allowed value is Persistent. When set, OVN Kubernetes assigned IP addresses will be persisted in an
+	// `ipamclaims.k8s.cni.cncf.io` object. These IP addresses will be reused by other pods if requested.
+	// Only supported by topology "Layer2" and "LocalNet" and when "subnets" are set.
+	// +optional
+	IPAMLifecycle NetworkIPAMLifecycle `json:"ipamLifecycle,omitempty"`
+
+	//TODO
+	//vlanIDs?
```

Comment:
right. VLAN would currently only make sense for localnet, but I guess there could be use-cases out there for primary layer2 UDNs.

Reply:
I will just remove it, but wanted to check if we ever plan to add this field to the api

Reply:
Oh we totally do.